virtcontainers/sandbox.go

```diff
@@ -1203,6 +1203,14 @@ func (s *Sandbox) Pause() error {
 		return err
 	}
 
+	//After the sandbox is paused, it's needed to stop its monitor,
+	//Otherwise, its monitors will receive timeout errors if it is
+	//paused for a long time, thus its monitor will not tell it's a
+	//crash caused timeout or just a paused timeout.
+	if s.monitor != nil {
+		s.monitor.stop()
```

Comment:
This looks like half the fix we need. If we stop the monitor when pausing the sandbox (literally pausing the VM), we need to re-init a watcher when resuming the VM, don't you think ?
I'd prefer to merge a PR solving the whole picture here.

Reply:
Hi, Sboeuf

I think the sandbox pause caller should responsible for re-init a monitor/watcher after it resume the sandbox.

Reply:
I cannot think of a case where the caller would not want to re-enable the monitor after it resumes the sandbox, for this reason, it's better if we get this as part of the Kata API.
@bergwolf WDYT ?